File: usecases/complete_todo.py
# usecases/complete_todo.py
from interfaces.todo_repository import TodoRepositoryInterface

class CompleteTodoUseCase:
    def __init__(self, repo: TodoRepositoryInterface):
        self.repo = repo
        self.complete_todo = CompleteTodoUseCase(repo)

    def execute(self, todo_id: int):
        todo = self.repo.find_by_id(todo_id)
        todo.completed = True
        self.repo.update(todo)
        return todo

File: aaa-practic/v6/use_cases/complete_todo.rb
require_relative '../entities/todo'

class CompleteTodo
  def initialize(repository)
    @repository = repository
  end
  
  def execute(todo_id)
    todo = @repository.find_by_id(todo_id)
    raise StandardError, "Todo with ID #{todo_id} not found" unless todo
    
    todo.completed = true
    @repository.update(todo)
    todo
  end
end
